apps/nextjs/src/components/AppComponents/Chat/prompt-form.tsx

```diff
@@ -13,56 +13,101 @@ import { useLessonPlanTracking } from "@/lib/analytics/lessonPlanTrackingContext
 import { useEnterSubmit } from "@/lib/hooks/use-enter-submit";
 import { useSidebar } from "@/lib/hooks/use-sidebar";
 
-interface PromptProps extends Pick<UseChatHelpers, "input" | "setInput"> {
+import { AilaStreamingStatus } from "./Chat/hooks/useAilaStreamingStatus";
+
+export interface PromptFormProps
+  extends Pick<UseChatHelpers, "input" | "setInput"> {
   onSubmit: (value: string) => void;
-  isLoading: boolean;
   isEmptyScreen: boolean;
   placeholder?: string;
+  ailaStreamingStatus: AilaStreamingStatus;
 }
 
 export function PromptForm({
+  ailaStreamingStatus,
   onSubmit,
   input,
   setInput,
-  isLoading,
   isEmptyScreen,
   placeholder,
-}: Readonly<PromptProps>) {
+}: Readonly<PromptFormProps>) {
   const { formRef, onKeyDown } = useEnterSubmit();
-  const inputRef = React.useRef<HTMLTextAreaElement>(null);
+  const inputRef = useRef<HTMLTextAreaElement>(null);
   const lessonPlanTracking = useLessonPlanTracking();
+  const [queuedUserInput, setQueuedUserInput] = useState<string | null>(null);
+  const timeoutRef = useRef<number | null>(null);
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (inputRef.current) {
       inputRef.current.focus();
     }
   }, []);
+
   const sidebar = useSidebar();
+
+  const queueSubmission = useCallback(
+    (value: string) => {
+      setQueuedUserInput(value);
+      timeoutRef.current = window.setTimeout(() => {
+        setQueuedUserInput(null);
+      }, 10000);
```

Comment:
The goal of the timeoutRef isn't very clear to me. What would the user see if it times out? Maybe a commend would be useful to explain it and/or the reason for 10000ms

Reply:
Would the message still be in the input field, or would it have disappeared in that case?

Reply:
I believe I've set it to still be displayed, but the input box should then be disabled, then the moderation completes, then it empties.